apps/api/src/database/repositories/boost.repository.ts

```diff
@@ -0,0 +1,183 @@
+import { and, eq, sql } from "drizzle-orm";
+import { z } from "zod";
+
+import { db } from "@/database/db.js";
+import * as schema from "@/database/schema/voting/defs.js";
+
+export { BoostRepository, BoostChangeMetaSchema };
+export type { BoostDiffArgs, BoostDiffResult };
+
+const BoostChangeMetaSchema = z.object({
+  description: z.string().optional(),
+});
+type BoostChangeMeta = z.infer<typeof BoostChangeMetaSchema>;
+const DEFAULT_META: BoostChangeMeta = {};
+
+type BoostDiffArgs = {
+  wallet: string;
+  amount: bigint;
+  meta?: BoostChangeMeta;
+  idemKey?: string;
+};
+
+type BoostDiffResult =
+  | { changeId: string; balanceAfter: bigint; idemKey: string } // change applied
+  | { balance: bigint; idemKey: string }; // no changes
+
+class BoostRepository {
+  readonly #db: typeof db;
+
+  constructor(database: typeof db = db) {
+    this.#db = database;
+  }
+
+  increase(args: BoostDiffArgs): Promise<BoostDiffResult> {
+    const amount = args.amount;
+    if (amount < 0n) {
+      throw new Error("amount must be non-negative");
+    }
+    const idemKey = args.idemKey ?? crypto.randomUUID();
+    const wallet = args.wallet.toLowerCase();
```

Comment:
e.g., you (correctly) lowercase the wallet address here (like we do elsewhere in our codebase). but just as a safeguard, maybe it's time to handle this in PG?

Reply:
See https://github.com/recallnet/js-recall/pull/1132#discussion_r2310562790

Oh, idemKey is also Uint8Array / bytea.